ratis-server/src/main/java/org/apache/ratis/server/storage/SnapshotManager.java

```diff
@@ -51,6 +53,7 @@ public class SnapshotManager {
 
   private final RaftStorage storage;
   private final RaftPeerId selfId;
+  private MessageDigest digester;
```

Comment:
Use
```
  private final Supplier<MessageDigest> digester = JavaUtils.memoize(MD5Hash::getDigester);
```
Then, the code change below will be reduced.

Reply:
done